File: tactix/tactix-training/infernoenv.js
const puppeteer = require('puppeteer'); // Puppeteer for headless browser

(async () => {
    // Launch a headless browser
    const browser = await puppeteer.launch({ headless: true });
    const page = await browser.newPage();

    // Open your React app (assuming it's running locally)
    await page.goto('http://localhost:8080'); // Adjust this if your React app is hosted elsewhere

    // Wait for the React app to fully load and WebSocket to connect (use any visible element for this)
    // If your app is not visually rendering anything, wait for a hidden selector or a state change
    await page.waitForSelector('body'); // Wait until the body is loaded, or use a more appropriate selector

    // Let the app run automatically – no need to manually trigger any actions
    // The WebSocket connection and execution will start automatically based on your app's internal logic

    // You can wait for the execution to finish or just let it run as long as needed
    // For example, you can wait for the app to finish processing or wait for a fixed amount of time
    // await page.w(10000); // 10 seconds, adjust this as needed
    console.log("App is running indefinitely, waiting for events...");

    // Infinite loop to keep the script running indefinitely
    while (true) {
        // For older Puppeteer versions use waitFor(), for newer use page.waitForTimeout()
        if (page.waitForTimeout) {
            await page.waitForTimeout(5000);
        } else {
            // For older versions
            await new Promise(resolve => setTimeout(resolve, 5000));
        }
        
        console.log("Still waiting...");

        // You can add more interaction logic here based on conditions in your app
    }

    // Close the browser
    await browser.close();
})();

File: tactix/tactix-training/train-headless/infernoenv.py
import asyncio
from pyppeteer import launch

async def main():
    # Launch a headless browser
    browser = await launch(
        headless=True,
        args=[
            '--no-sandbox',
            '--disable-setuid-sandbox',
            '--disable-dev-shm-usage',
            '--disable-gpu',
            '--disable-software-rasterizer'
        ]
    )
    page = await browser.newPage()

    # Open your React app (assuming it's running locally)
    await page.goto('http://localhost:8080')  # Adjust this if your React app is hosted elsewhere

    # Wait for the React app to fully load and WebSocket to connect (use any visible element for this)
    await page.waitForSelector('body')  # Wait until the body is loaded, or use a more appropriate selector

    print("App is running indefinitely, waiting for events...")

    # Infinite loop to keep the script running indefinitely
    while True:
        # For older Pyppeteer versions use waitFor(), for newer use page.waitForTimeout()
        try:
            await page.waitFor(5000)  # 5 seconds, adjust this as needed
        except Exception as e:
            print(f"Error: {e}")
        
        print("Still waiting...")

    # Close the browser
    await browser.close()

# Run the event loop
asyncio.get_event_loop().run_until_complete(main())